lambda/shared/src/main/scala/feral/lambda/IOLambda.scala

```diff
@@ -30,15 +31,13 @@ abstract class IOLambda[Event, Result](
     with IOSetup {
 
   final type Setup = (Event, Context[IO]) => IO[Option[Result]]
-  final override protected def setup: Resource[IO, Setup] =
-    handler.map { handler => (event, context) =>
-      for {
-        event <- IOLocal(event)
-        context <- IOLocal(context)
-        env = LambdaEnv.ioLambdaEnv(event, context)
-        result <- handler(env)
-      } yield result
-    }
+  final override protected def setup: Resource[IO, Setup] = for {
+    handler <- handler
+    localEvent <- IOLocal[Event](null.asInstanceOf[Event]).toResource
+    localContext <- IOLocal[Context[IO]](null).toResource
+    env = LambdaEnv.ioLambdaEnv(localEvent, localContext)
+    result = handler(env)
+  } yield { localEvent.set(_) *> localContext.set(_) *> result }
```

Comment:
This looks a bit uglier and is less safe, but all optimizations are 😉 this lets us avoid building the user-provided handler on every invocation, since acquiring the handler resource actually returns a function `LambdaEnv => IO` and not the final `IO`.

Reply:
is that going to be more efficient if we'll be doing two `.set`s on these locals in every request anyway?

Reply:
Hmm, why would using `.set` be inefficient?

So if your function `LambdaEnv => IO` is something like:
`env => Middleware(Middleware(Middleware(handler)))`
basically I'm trying to avoid invoking that method on every request.